include/deal.II/numerics/error_estimator.templates.h

```diff
@@ -1160,30 +1160,23 @@ KellyErrorEstimator<dim, spacedim>::estimate(
   const types::material_id  material_id,
   const Strategy            strategy)
 {
-#ifdef DEAL_II_WITH_P4EST
-  if (dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim> *>(
-        &dof_handler.get_triangulation()) != nullptr)
-    Assert((subdomain_id_ == numbers::invalid_subdomain_id) ||
-             (subdomain_id_ ==
-              dynamic_cast<
-                const parallel::distributed::Triangulation<dim, spacedim> &>(
-                dof_handler.get_triangulation())
-                .locally_owned_subdomain()),
-           ExcMessage(
-             "For parallel distributed triangulations, the only "
-             "valid subdomain_id that can be passed here is the "
-             "one that corresponds to the locally owned subdomain id."));
-
-  const types::subdomain_id subdomain_id =
-    ((dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim> *>(
-        &dof_handler.get_triangulation()) != nullptr) ?
-       dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim>
-                      &>(dof_handler.get_triangulation())
-         .locally_owned_subdomain() :
-       subdomain_id_);
-#else
-  const types::subdomain_id subdomain_id = subdomain_id_;
-#endif
+  types::subdomain_id subdomain_id = numbers::invalid_subdomain_id;
+  if (const auto *triangulation = dynamic_cast<
+        const parallel::DistributedTriangulationBase<dim, spacedim> *>(
+        &dof_handler.get_triangulation()))
+    {
+      Assert((subdomain_id == numbers::invalid_subdomain_id) ||
```

Comment:
Isn't this check redundant as you set the variable to this specific value above?

Reply:
I checked the wrong thing - those should be for the input parameter `subdomain_id_` and not `subdomain_id`.